web/views.py

```diff
@@ -40,7 +42,29 @@ def project_detail(request,project_id):
 	project = get_object_or_404(Project, project_id=project_id)
 	return render(request, 'pages/project_detail.html', {'project': project})
 
+def project_new(request):
+	if request.method == "POST":
+		projectform = ProjectForm (request.POST)
+		if projectform.is_valid():
+			project = projectform.save()
+			return redirect('web.views.project_detail', project_id=project.project_id)
+	else:
+		projectform = ProjectForm()
+
+	return render(request, 'pages/project_edit.html', {'projectform': projectform})
+
+def project_edit(request, project_id):
+	project = get_object_or_404(Project, project_id=project_id)
+	if request.method == "POST":
+		projectform = ProjectForm(request.POST, instance=project)
+		if projectform.is_valid():
+			project = projectform.save()
+			return redirect('web.views.project_detail', project_id=project.project_id)
+	else:
+		projectform = ProjectForm(instance=project)
+
+	return render(request, 'pages/project_edit.html', {'projectform': projectform})
+
 def logout(request):
     auth_logout(request)
     return redirect('/')
-
```

Comment:
Why did you delete a new line at the end?

Ah I know, to annoy @goranche.... 

Please do add new line to the end of the file.